script.js

```diff
@@ -1,12 +1,275 @@
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  for (let i = 0; i < episodeList.length; i++) {
+    let card = createCardForEpisodes(episodeList[i]);
+    parent.appendChild(card);
+  }
+  createDropdown(episodeList);
+}
+
+function createCardForEpisodes(episode) {
+  card = document.createElement("div");
+  card.className = "card";
+  card.style.textAlign = "center";
+  card.appendChild(createTitleEpisode(episode));
+  let episodeCardImage
+  if (episode.image) {
+    episodeCardImage = createImage(episode.image.medium);  
+  } else {
+    episodeCardImage= createImage("https://cdn.hswstatic.com/gif/simpsons-31.jpg");
+  }
+  card.appendChild(episodeCardImage);
+  card.appendChild(createSummary(episode.summary));
+  return card;
+}
+
+function createTitleEpisode(episode) {
+  let title = document.createElement("h1");
+  title.style.margin = "0 30px 40px 30px";
+  title.style.fontSize = "x-large";
+  title.innerText = `${episode.name} - ${formatEpisodeNum(
+    episode.season,
+    episode.number
+  )}`;
+  return title;
+}
+
+function formatEpisodeNum(season, number) {
+  return `S${season.toString().padStart(2, "0")}E${number
+    .toString()
+    .padStart(2, "0")}`;
+}
+
+// ------------- create episode and show Page  ---------------
+
+function createSummary(summary) {
+  let summaryElement = document.createElement("p");
+  summaryElement.style.margin = "30px 30px 20px 30px";
+  summaryElement.style.textAlign = "justify";
+  summaryElement.innerHTML = summary;
+  return summaryElement;
+}
+
+function createImage(imageSrc) {
+  let imageElement = document.createElement("img");
+  imageElement.src = imageSrc;
+  return imageElement;
+}
+
+// ------------- Create Episode Dropdown on onload ------------
+
+function createDropdown(allEpisodes) {
+  episodeSelector.innerHTML = '';
+  createSelectAllEpisodesOption()
+  for (let i = 0; i < allEpisodes.length; i++) {
+    let optionTitle = document.createElement("option");
+    episodeSelector.appendChild(optionTitle);
+    optionTitle.innerText = `${formatEpisodeNum(
+      allEpisodes[i].season,
+      allEpisodes[i].number
+    )} - ${allEpisodes[i].name}`;
+  }
+}
+
+function createSelectAllEpisodesOption() {
+  let optionTitle = document.createElement("option");
+  episodeSelector.appendChild(optionTitle);
+  optionTitle.innerText = "Select All Episodes";
+  
+
+   optionTitle.addEventListener("change", setup);
+}
+
+// ------------ Episode Selector -----------------
+
+episodeSelector.addEventListener("change", function (event) {  
+ 
+  let episodeOption = event.target.value;
+  episodeOption = episodeOption.slice(0, 6);
+  let episodes = document.querySelectorAll(".card");
+
+  let newArrayOfEpisodes = Array.from(episodes);
+  newArrayOfEpisodes.forEach(function (card) {
+    if (!card.innerText.includes(episodeOption)) {
+      card.style.display = "none";
+    } else if (card.innerText.includes(episodeOption)) {
+      card.style.display = "block";
```

Comment:
u could apply 'select all episodes' here so when u select it all episodes displayed block 
else if (card.innerText.includes(episodeOption) || episodeOption === "Select All Episodes")

Reply:
Very good suggestion. You could check first if selected option equals to "Select All Episodes", or even give it a shorter value, and display all episodes, otherwise display selected episode.